_includes/tutorials/fk-joins/kstreams/markup/dev/make-joiner.adoc

```diff
@@ -0,0 +1,9 @@
+For the ValueJoiner class, create the following file at `src/main/java/io/confluent/developer/MusicInterestJoiner.java`.
+
+When you join two tables in a relational database, by default you get a new table containing all of the columns of the left table plus all of the columns of the right table. When you join a table and a table, you get a new table, but you must be explicit about the value of that table—the combination between the value in the table and the assocaited value in the table. The `https://kafka.apache.org/{{ site.ak_javadoc_version }}/javadoc/org/apache/kafka/streams/kstream/ValueJoiner.html[ValueJoiner]` interface in the Streams API does this work. The single `apply()` method takes the "left" table and the "right" table values as parameters, and returns the value of the joined table as output. (Their keys are not a part of the equation, because they are equal by definition and do not change in the result.) As you can see here, this is just a matter of creating a `MusicInterest` object and populating it with the relevant fields of the input album and track purchase.
```

Comment:
> When you join two tables in a relational database, by default you get a new table containing all of the columns of the left table plus all of the columns of the right table. When you join a table and a table, you get a new table, but you must be explicit about the value of that table—the combination between the value in the table and the assocaited value in the table.

This was difficult to follow, maybe try a revision?

Reply:
ack